lib/pages/merge.js

```diff
@@ -15,68 +20,127 @@ module.exports = React.createClass({displayName: 'MergePage',
         });
     },
 
-    squashMergeWithRewrite: function(event) {
-        var pr = this.state.pr_info.pr;
-        var commitMessage = pr.getTitle();
-        if (pr.getBody()) {
-            commitMessage += "\n\n" + pr.getBody();
-        }
-        this.props.controller.squashMergeWithRewrite(pr, commitMessage);
+    getInitialState: function() {
+        // TODO: Also do this when re-selecting the tab
+        return {
+            flowStage: "selectMergeKind",
+            mergeKind: null,
+            commitMessage: null
+        };
     },
 
-    squashMergeWithoutRewrite: function(event) {
+    promptForCommitMessage: function(event) {
         var pr = this.state.pr_info.pr;
+
         var commitMessage = pr.getTitle();
         if (pr.getBody()) {
             commitMessage += "\n\n" + pr.getBody();
         }
-        commitMessage += "\n\nSquash-merged from pull request #" + pr.getId() +
-            " from " + pr.getSource().getUser().name + "/" + pr.getSource().getRef();
-        this.props.controller.squashMergeWithoutRewrite(pr, commitMessage);
+
+        if (event.target.textContent === squashMerge) {
+            commitMessage += "\n\nSquash-merged from pull request #" + pr.getId();
+        }
+        else if (event.target.textContent === mergeWithoutSquash) {
+            commitMessage = "Merge pull request #" + pr.getId() + " from " +
+                pr.getSource().getUser().name + "/" + pr.getSource().getRef();
```

Comment:
Given we're using ES6 features, it *may* be clearer to use template strings here. Could be fun!

https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/template_strings